src/main/kotlin/org/rust/lang/core/psi/RsPsiManager.kt

```diff
@@ -60,30 +60,28 @@ class RsPsiManagerImpl(val project: Project) : ProjectComponent, RsPsiManager {
         })
     }
 
-    inner class CacheInvalidator : PsiTreeChangeAdapter() {
-        override fun beforeChildRemoval(event: PsiTreeChangeEvent) = onPsiChange(event)
-        override fun beforeChildReplacement(event: PsiTreeChangeEvent) = onPsiChange(event, event.oldChild)
-        override fun childReplaced(event: PsiTreeChangeEvent) = onPsiChange(event)
-        override fun childAdded(event: PsiTreeChangeEvent) = onPsiChange(event)
-        override fun childrenChanged(event: PsiTreeChangeEvent) = onPsiChange(event)
-        override fun childMoved(event: PsiTreeChangeEvent) = onPsiChange(event)
-        override fun propertyChanged(event: PsiTreeChangeEvent) = onPsiChange(event)
-
-        private fun onPsiChange(event: PsiTreeChangeEvent, child: PsiElement? = event.child) {
-            // `GenericChange` event means that "something changed in the file" and sends
-            // after all events for concrete PSI changes in a file.
-            // We handle more concrete events and so should ignore generic event
-            if (event is PsiTreeChangeEventImpl && event.isGenericChange) return
+    inner class CacheInvalidator : RsPsiTreeChangeAdapter() {
+        override fun handleEvent(event: RsPsiTreeChangeEvent) {
+            val child = when (event) {
```

Comment:
It seems that `element` name or something similar is more suitable here because sometimes we assign `event.parent` to `child` variable and it can be misleading